### PR TITLE
Private Data support

### PR DESCRIPTION
**Pull request by Hejwo**

### CODE REVIEW

File: e2e/__snapshots__/network-01-simple.test.js.snap [hunk 1]
```diff
@@ -1221,3 +1245,338 @@ Array [
   "e2e/__tmp__/network-01-simple/fabrica-target/fabric-docker/scripts/cli/channel_fns.sh",
 ]
 `;
+
+exports[`network-01-simple should extend config properly 1`] = `
+"Fabrica is powered by :
```
Comment (dzikowski) [inline e2e/__snapshots__/network-01-simple.test.js.snap:1250]: tests for extend config were added

File: samples/fabricaConfig-1org-1channel-1chaincode.json [hunk 2]
```diff
@@ -61,7 +61,15 @@
       "channel": "channel1",
       "init": "{\"Args\":[]}",
       "endorsement": "AND ('Org1MSP.member')",
-      "directory": "./chaincodes/chaincode-kv-node"
+      "directory": "./chaincodes/chaincode-kv-node",
+      "privateData": [
```
Comment (Hejwo) [inline samples/fabricaConfig-1org-1channel-1chaincode.json:65]: We copy that file to user during `fabrica init` so it should be as concise and simple as possible.

File: docs/schema.json [hunk 3]
```diff
@@ -64,9 +67,11 @@
               ],
               "default": "info"
             }
-          }
+          },
+          "additionalProperties": false
```
[the remark this answers is not in the document]
Comment (dzikowski) [reply docs/schema.json:71]: it forbids to add properties that are not included in the schema

File: docs/sample.json [hunk 4]
```diff
@@ -61,7 +61,15 @@
...
       "endorsement": "AND ('Org1MSP.member')",
-      "directory": "./chaincode1"
+      "directory": "./chaincode1",
+      "privateData": [
+        {
+          "name": "privateCollectionOrg1",
+          "orgNames": [
+            "Org1"
+          ]
+        }
```
[the remark this answers is not in the document]
Comment (dzikowski) [reply docs/sample.json:71]: TODO: test 1.3, consider `memberOnlyWrite` in #156

File: docs/schema.json [hunk 5]
```diff
@@ -344,7 +356,6 @@
           "name": "my-channel2"
         }
       ],
-      "additionalItems": true,
```
Comment (dzikowski) [inline docs/schema.json:347]: it should probably be reverted

File: docs/schema.json [hunk 6]
```diff
@@ -497,9 +507,55 @@
...
+              "required": [
+                "name",
+                "orgNames"
+              ],
+              "properties": {
+                "name": {
+                  "$id": "#/properties/chaincodes/items/properties/privateData/items/properties/name",
+                  "title": "Collection name",
+                  "type": "string",
+                  "pattern": "^[A-Za-z0-9]+$"
```
[the remark this answers is not in the document]
Comment (dzikowski) [reply docs/schema.json:528]: TBD

File: e2e-network/TEST_CASES.md [hunk 7]
```diff
@@ -12,6 +12,7 @@
 | Channels                  | 1     | 2     |
 | Node chaincode            | yes   | yes   |
 | Node chaincode upgrade    | yes   | yes   |
+| Private data              | yes   | no    |
```
[the remark this answers is not in the document]
Comment (dzikowski) [reply e2e-network/TEST_CASES.md:15]: #156

File: samples/chaincodes/chaincode-kv-node/index.js [hunk 8]
```diff
@@ -19,6 +20,56 @@ class KVContract extends Contract {
     if (!buffer || !buffer.length) return { error: 'NOT_FOUND' };
     return { success: buffer.toString() };
   }
+
```
[the remark this answers is not in the document]
Comment (dzikowski) [reply samples/chaincodes/chaincode-kv-node/index.js:23]: `getHistory` should be removed

TODO: create new JSON for config with private data (avoid private data in fabrica init)

File: docs/schema.json (hunk 5 above)
Comment (dzikowski) [reply docs/schema.json:347]: nope, it should not https://json-schema.org/understanding-json-schema/reference/array.html#list-validation

File: docs/schema.json (hunk 6 above)
[the remark this answers is not in the document]
Comment (dzikowski) [reply docs/schema.json:528]: from logs: `A valid collection name follows the pattern: [A-Za-z0-9_-]+`

fixed

File: samples/chaincodes/collections_config.json [hunk 9]
```diff
@@ -0,0 +1,10 @@
+[
```
[the remark this answers is not in the document]
Comment (dzikowski) [reply samples/chaincodes/collections_config.json:1]: removed

File: docs/sample.json [hunk 10]
```diff
@@ -1,5 +1,5 @@
 {
-  "$schema": "https://github.com/softwaremill/fabrica/releases/download/0.0.1/schema.json",
+  "$schema": "https://github.com/softwaremill/fabrica/releases/download/0.1.0-unstable/schema.json",
```
Comment (Hejwo) [inline docs/sample.json:2]: :+1:

**Pull request merged**